### PR TITLE
Api2.0 fixes

### PR DESCRIPTION
**Pull request by nkogteva**
### Details:
 - *Fix version printed for new api: pointer is invalid in user application*
 - *Fix case when layout set for input with number of dimensions > 5 (`Layout::BLOCKED`)*: it set to `[N,L,O,C,K,E,D]` for incompatible shape
 - *Add method for setting task/callback executors in compiled model (required for nvidia plugin)*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/inference/src/dev/preprocessing/preprocessing.cpp [hunk 1]
```diff
@@ -33,9 +33,11 @@ bool ov::pass::AddPreprocessing::run_on_model(const std::shared_ptr<ov::Model>&
         preproc.input(i).tensor().set_element_type(
             InferenceEngine::details::convertPrecision(input_info->getPrecision()));
 
-        std::stringstream stream;
-        stream << input_info->getLayout();
-        preproc.input(i).tensor().set_layout(ov::Layout{stream.str()});
+        if (input_info->getLayout() != InferenceEngine::Layout::BLOCKED) {
```
Comment (vladimir-paramuzov) [inline src/inference/src/dev/preprocessing/preprocessing.cpp:36]: Don't we need similar handling for `Layout::SCALAR` and `Layout::ANY`?

Comment (nkogteva) [reply src/inference/src/dev/preprocessing/preprocessing.cpp:36]: @vladimir-paramuzov `ANY` can't arise there 

https://github.com/openvinotoolkit/openvino/blob/a4314faf292212d5b21544444b55c2eaac8af807/src/inference/src/ie_layouts.cpp#LL172C1-L189C2

Comment (nkogteva) [reply src/inference/src/dev/preprocessing/preprocessing.cpp]: `SCALAR` was added

File: src/inference/src/dev/preprocessing/preprocessing.cpp [hunk 2]
```diff
@@ -33,9 +33,12 @@ bool ov::pass::AddPreprocessing::run_on_model(const std::shared_ptr<ov::Model>&
         preproc.input(i).tensor().set_element_type(
             InferenceEngine::details::convertPrecision(input_info->getPrecision()));
 
-        std::stringstream stream;
-        stream << input_info->getLayout();
-        preproc.input(i).tensor().set_layout(ov::Layout{stream.str()});
+        if (input_info->getLayout() != InferenceEngine::Layout::BLOCKED &&
```
Comment (ilyachur) [inline src/inference/src/dev/preprocessing/preprocessing.cpp:36]: What is about "ANY"?

Comment (nkogteva) [reply src/inference/src/dev/preprocessing/preprocessing.cpp:36]: @ilyachur see my answer above: `ANY` can't arise there

https://github.com/openvinotoolkit/openvino/blob/a4314faf292212d5b21544444b55c2eaac8af807/src/inference/src/ie_layouts.cpp#LL172C1-L189C2

**Comment by nkogteva** [comment]
> > Clean up extensions in FrontEnd destructor
> 
> Should we port it to 2022.3 LTS?

Not sure how to reproduce problem on LTS.

**Review by ilyachur — APPROVED** [review]

**Pull request merged**